samples/glean/src/androidTest/java/org/mozilla/samples/glean/pings/BaselinePingTest.kt

```diff
@@ -0,0 +1,134 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.samples.glean.pings
+
+import android.content.Context
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.platform.app.InstrumentationRegistry
+import androidx.test.rule.ActivityTestRule
+import org.junit.Assert.assertEquals
+
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mozilla.samples.glean.MainActivity
+import org.mozilla.samples.glean.getPingServer
+import androidx.test.uiautomator.UiDevice
+import androidx.work.WorkInfo
+import androidx.work.WorkManager
+import androidx.work.testing.WorkManagerTestInitHelper
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.withTimeout
+import mozilla.components.service.glean.config.Configuration
+import mozilla.components.service.glean.testing.GleanTestRule
+import org.json.JSONObject
+import org.junit.Assert.assertFalse
+import org.junit.Assert.assertTrue
+import org.junit.Before
+import org.mozilla.samples.glean.getPingServerAddress
+import java.util.concurrent.TimeUnit
+
+@RunWith(AndroidJUnit4::class)
+class BaselinePingTest {
+    @get:Rule
+    val activityRule: ActivityTestRule<MainActivity> = ActivityTestRule(MainActivity::class.java)
+
+    @get:Rule
+    val gleanRule = GleanTestRule(
+        ApplicationProvider.getApplicationContext(),
+        Configuration(serverEndpoint = getPingServerAddress())
+    )
+
+    @Before
+    fun clearWorkManager() {
+        val context = ApplicationProvider.getApplicationContext<Context>()
+        WorkManagerTestInitHelper.initializeTestWorkManager(context)
+    }
+
+    private fun waitForPingContent(
+        pingName: String,
+        maxAttempts: Int = 3
+    ): JSONObject?
+    {
+        val server = getPingServer()
+
+        var attempts = 0
+        do {
+            attempts += 1
+            val request = server.takeRequest(20L, TimeUnit.SECONDS)
+            val docType = request.path.split("/")[3]
+            if (pingName == docType) {
+                return JSONObject(request.body.readUtf8())
+            }
+        } while (attempts < maxAttempts)
+
+        return null
+    }
+
+    /**
+     * Sadly, the WorkManager still requires us to manually trigger the upload job.
+     * This function goes through all the active jobs by Glean (there should only be
+     * one!) and triggers them.
```

Comment:
Would there be any benefit in asserting that it's just one?

Reply:
I don't think so, since we're explicitly using a `WorkManager` API that should take care of that (`enqueueUniqueWork`). We'd be testing that API :)